### PR TITLE
[Scala] File upload using new additional router API

### PR DESCRIPTION
**Pull request by renatocaval**

### CODE REVIEW

File: file-upload/file-upload-scala-sbt/fileupload-impl/src/main/scala/com/example/fileupload/impl/FileUploadRouter.scala [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import scala.concurrent.{ ExecutionContext, Future }
+import scala.concurrent.Future
+
+class FileUploadRouter(action: DefaultActionBuilder,
+                       parser: PlayBodyParsers,
+                       implicit val exCtx: ExecutionContext) {
+
+  // A Play FilePartHandler[T] creates an Accumulator (similar to Akka Stream's Sinks)
+  // for each FileInfo in the multipart request.
+  private def fileHandler: FilePartHandler[File] = {
```
[the remark this answers is not in the document]
Comment (marcospereira) [reply file-upload/file-upload-scala-sbt/fileupload-impl/src/main/scala/com/example/fileupload/impl/FileUploadRouter.scala]: By the way, https://github.com/playframework/playframework/issues/9066.

[the remark this answers is not in the document]
Comment (renatocaval) [reply file-upload/file-upload-scala-sbt/fileupload-impl/src/main/scala/com/example/fileupload/impl/FileUploadRouter.scala]: I don't know why it was built like that. I'm just porting the exiting code. I'm happy to improve it later on another PR.

File: file-upload/file-upload-scala-sbt/fileupload-impl/src/main/scala/com/example/fileupload/impl/FileUploadRouter.scala [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        // create a temp file in the `target` folder
+        val f = new java.io.File("./target/file-upload-data/uploads", UUID.randomUUID().toString).getAbsoluteFile
+        // make sure the sub-folders inside `target` exist.
+        f.getParentFile.mkdirs()
+        f
+      }
+      val sink: Sink[ByteString, Future[IOResult]] = FileIO.toPath(tempFile.toPath)
+      val acc: Accumulator[ByteString, IOResult] = Accumulator(sink)
+      acc.map {
+        case akka.stream.IOResult(_, _) =>
```
[the remark this answers is not in the document]
Comment (renatocaval) [reply file-upload/file-upload-scala-sbt/fileupload-impl/src/main/scala/com/example/fileupload/impl/FileUploadRouter.scala]: Same as above, I'm just porting the code as is. We can improve it in another PR.

**Comment by ignasi35** [comment]
Just noticed something (after reviewing #68):

 * please update `sbt.version` to `1.2.8` on `build.props`

**Pull request merged**